extract/spacy_model.py

```diff
@@ -0,0 +1,101 @@
+import sys
+
+sys.path.append("../")
+import json
+import random
+import spacy
+import pandas as pd
+import os
+from lib.path import data_file_path
+
+# #  import labeled data that has been exported from doccano
+labeled_data = []
+
+with open(
+    data_file_path("raw/new_orleans_csc/appeal_hearing_pdfs/training_data/ayyub.jsonl")
+) as read_file:
+    for line in read_file:
+        data = json.loads(line)
+        labeled_data.append(data)
+
+
+TRAINING_DATA = []
+for entry in labeled_data:
+    entities = []
+    for e in entry["label"]:
+        entities.append((e[0], e[1], e[2]))
+    spacy_entry = (entry["data"], {"entities": entities})
+    TRAINING_DATA.append(spacy_entry)
+
+
+#  load a spacy model OR
+# nlp = spacy.load("C:/Users/ayyubi/extraction/data/raw/new_orleans_csc/training/json/nopd.model")
+
+# #  train a new spacy model
+nlp = spacy.blank("en")
+ner = nlp.create_pipe("ner")
+nlp.add_pipe(ner)
+ner.add_label("docket_number")
+ner.add_label("accused_name")
+ner.add_label("incident_summary")
+ner.add_label("appeal_disposition")
+ner.add_label("allegation")
+ner.add_label("initial_action")
+ner.add_label("notification_of_action_date")
+ner.add_label("incident_date")
+ner.add_label("nopd_incident_report_number")
+ner.add_label("attorney_name")
+ner.add_label("appeal_date")
+
+nlp.begin_training()
+for itn in range(25):
+    random.shuffle(TRAINING_DATA)
+    losses = {}
+    for batch in spacy.util.minibatch(TRAINING_DATA, size=2):
+        texts = [text for text, entities in batch]
+        annotations = [entities for text, entities in batch]
+        nlp.update(texts, annotations, losses=losses, drop=0.3)
+    print(losses)
```

Comment:
Try to put all configs in a `config.cfg` file, remove this script and invoke the training via `python -m spacy train config.cfg` instead. Convert training and evaluation data to spacy format using existing command line or if it's not available, write your own script and invoke them as necessary.

Reply:
[Prodigy](https://prodi.gy/docs/recipes#data-to-spacy) seems to do the same thing as Doccano, but it can output data directly in Spacy format. Any reason why you did not use Prodigy?

Reply:
Got it. Thanks.

Reply:
Decided not to buy the Prodigy subscription because the team was already using Doccano in Tarak's workflow.

Reply:
I see